src/playback/internal/playbackcontroller.cpp

```diff
@@ -194,45 +202,53 @@ void PlaybackController::playElement(const notation::Element* element)
         return;
     }
 
-    notationPlayback()->playElementMidiData(element);
+    masterNotationMidiData()->triggerElementMidiData(element);
 }
 
 INotationPlaybackPtr PlaybackController::notationPlayback() const
 {
     return m_notation ? m_notation->playback() : nullptr;
 }
 
+IMasterNotationMidiDataPtr PlaybackController::masterNotationMidiData() const
+{
+    return m_masterNotation ? m_masterNotation->midiData() : nullptr;
+}
+
+INotationPartsPtr PlaybackController::masterNotationParts() const
+{
+    return m_masterNotation ? m_masterNotation->parts() : nullptr;
+}
+
 INotationSelectionPtr PlaybackController::selection() const
 {
     return m_notation ? m_notation->interaction()->selection() : nullptr;
 }
 
 void PlaybackController::onNotationChanged()
 {
-    playback()->player()->stop(m_currentSequenceId);
-    playback()->player()->playbackPositionMsecs().resetOnReceive(this);
-
+    m_masterNotation = globalContext()->currentMasterNotation();
     m_notation = globalContext()->currentNotation();
-    if (!m_notation) {
-        return;
-    }
-
-    project::INotationProjectPtr project = globalContext()->currentProject();
-    IF_ASSERT_FAILED(project) {
+    if (!m_notation || !m_masterNotation) {
         return;
     }
 
-    io::path notationKey = project->path();
+    INotationPartsPtr notationParts = m_notation->parts();
 
-    auto search = m_sequenceIdMap.find(notationKey.toStdString());
+    for (const Part* part : m_masterNotation->parts()->partList()) {
```

Comment:
better to move this code to a separate method like muteNonexistentParts, because it is not clear what is happening here